### PR TITLE
Interactive mode for user-input

### PR DESCRIPTION
**Pull request by surajssd**
Added some checks on data received from docker-compose file, viz: builds and service ports

### CODE REVIEW

**Comment by pradeepto** [comment]
Is this PR related to https://github.com/redhat-developer/henge/issues/4 ? If so, we should say so and that issue needs to be assigned to @surajssd

**Comment by surajssd** [comment]
So with this the focus is to update the parsed docker-compose file's internal data structure which is `p.Config`

**Comment by rtnpro** [comment]
@surajssd Your approach to accept input from the user looks good to me. I found some good read here: http://thenewstack.io/cli-command-line-programming-with-go/. May be, you want to have a look.

File: henge.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ func main() {
 		os.Exit(1)
 	}
 
-	err = transformers.Transform(*provider, flag.Args()[0:]...)
+	err = transformers.Transform(*provider, *interactive, flag.Args()[0:]...)
```
Comment (surajssd) [inline henge.go]: better way to pass arguments around?

File: pkg/transformers/openshift/lookup.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package openshift
```
Comment (surajssd) [inline pkg/transformers/openshift/lookup.go]: are we utilizing this file?

File: henge.go (hunk 1 above)
Comment (kadel) [inline henge.go]: I think this is ok for now. 
This is going to be better after we switch to cobra.

File: pkg/transformers/openshift/lookup.go (hunk 2 above)
Comment (kadel) [inline pkg/transformers/openshift/lookup.go]: I don't think so.
I think that we can delete this.

File: henge.go (hunk 1 above)
Comment (surajssd) [inline henge.go]: ACK

File: pkg/transformers/openshift/generate.go [hunk 3]
```diff
@@ -0,0 +1,569 @@
+package openshift
```
Comment (surajssd) [inline pkg/transformers/openshift/generate.go]: What about this file we add it in this PR?

Comment (surajssd) [inline pkg/transformers/openshift/generate.go]: As discussed with @rtnpro on IRC let this be in, in this PR.

**Comment by surajssd** [comment]
currently it looks like this: 

```bash
11:29 $ cat docker-compose-build.yml 
version: "2"

services:
  mariadb:
    image: centos/mariadb
    environment:
      MYSQL_ROOT_PASSWORD: etherpad
      MYSQL_DATABASE: etherpad
      MYSQL_PASSWORD: etherpad
      MYSQL_USER: etherpad

  etherpad:
    build: "./myphp"
    ports:
      - "80:9001"
    depends_on:
      - mariadb
    environment:
      DB_HOST: mariadb
      DB_DBID: etherpad
      DB_PASS: etherpad
      DB_USER: etherpad
✔ ~/compose-lib/etherpad [master|✚ 2…6] 
08:26 $ ./henge -target openshift -interactive docker-compose-build.yml 
[mariadb] No ports defined to send traffic to, no provider service will be created. Do you want to create a service? y/[n]: y
[mariadb] Enter ports: 3306
[etherpad] Do you want to use "/home/vagrant/compose-lib/etherpad/myphp"'s git origin as source? [y]/n: n
[etherpad] Enter application image name: centos/etherpad
apiVersion: v1
items:
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: etherpad
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: centos/etherpad
      from:
        kind: DockerImage
        name: centos/etherpad
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: etherpad
  spec:
    replicas: 1
    selector:
      deploymentconfig: etherpad
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: etherpad
      spec:
        containers:
        - env:
          - name: DB_DBID
            value: etherpad
          - name: DB_HOST
            value: mariadb
          - name: DB_PASS
            value: etherpad
          - name: DB_USER
            value: etherpad
          image: centos/etherpad
          name: etherpad
          ports:
          - containerPort: 9001
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - etherpad
        from:
          kind: ImageStreamTag
          name: etherpad:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: ImageStream
  metadata:
    creationTimestamp: null
    name: mariadb
  spec:
    tags:
    - annotations:
        openshift.io/imported-from: centos/mariadb
      from:
        kind: DockerImage
        name: centos/mariadb
      generation: null
      importPolicy: {}
      name: latest
  status:
    dockerImageRepository: ""
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    creationTimestamp: null
    name: mariadb
  spec:
    replicas: 1
    selector:
      deploymentconfig: mariadb
    strategy:
      resources: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          deploymentconfig: mariadb
      spec:
        containers:
        - env:
          - name: MYSQL_DATABASE
            value: etherpad
          - name: MYSQL_PASSWORD
            value: etherpad
          - name: MYSQL_ROOT_PASSWORD
            value: etherpad
          - name: MYSQL_USER
            value: etherpad
          image: centos/mariadb
          name: mariadb
          ports:
          - containerPort: 3306
          resources: {}
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - mariadb
        from:
          kind: ImageStreamTag
          name: mariadb:latest
      type: ImageChange
  status: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: etherpad
  spec:
    ports:
    - name: 9001-tcp
      port: 9001
      targetPort: 9001
    selector:
      deploymentconfig: etherpad
  status:
    loadBalancer: {}
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: null
    name: mariadb
  spec:
    ports:
    - name: 3306-tcp
      port: 3306
      targetPort: 3306
    selector:
      deploymentconfig: mariadb
  status:
    loadBalancer: {}
kind: List
metadata: {}
```

**Pull request merged**